### PR TITLE
Adding Create VM and SSH & aksQuickStart tutorials along with additional error checking 

### PR DESCRIPTION
**Pull request by jasonmesser7**
I found in development if a code block which did something along the lines of

```yaml
    cat <<EOF > azure-vote.yaml
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: azure-vote-back
      
    EOF
```
That the command when it was given to the REPL would hang as the EOF token would not be in the right place i.e immediately after the last line. I added a function which checks for any commands with EOF and removes unnecessary spaces between the EOF token and the last line.

This commit also adds two tutorials - Create VM and SSH and aksQuickStart

### CODE REVIEW

File: demoScripts/aksQuickStart.md
Comment (rgardler-msft) [inline demoScripts/aksQuickStart.md:357]: The timeout here is for automated testing, right?

I wonder if we should implement a "special command" parser in Innovation Engine. the original SimDem code could do this, it would look for commands that needed to be handled differently in different environments and rewrite them in test. Here we could look for the `--watch` and automatically add the `timeout 30` if not already present.

Doesn't need to be done in this doc but if it is a viable idea we should create an issue for this.

File: demoScripts/aksQuickStart.md
Comment (rgardler-msft) [inline demoScripts/aksQuickStart.md:373]: We need to test this otherwise there is no guarantee that the application was deployed correctly. 

A downside of this approach is the `timeout` approach is that we would need a separate command to get the IP number for testing. I do, however, prefer the readability of this approach, especially if are further able to simplify with the "special command" approach.

What we could do is something like:

It is often useful to have the IP address in an environment variable for later reuse. You can do this by running:

```
export QUICKSTART_AKS_IP=...
```

File: demoScripts/aksQuickStart.md
Comment (rgardler-msft) [inline demoScripts/aksQuickStart.md:386]: What about the second RG that it created, will that be cleaned up too?

File: demoScripts/createVMAndSSH.md
Comment (rgardler-msft) [inline demoScripts/createVMAndSSH.md:8]: Meaningful names please, for both the environment variable and the value, e.g.

```
export QUICKSTART_VM_RESOURCE_GROUP=vmWishSSHQuickstart
```

Comment (rgardler-msft) [reply demoScripts/createVMAndSSH.md:8]: Additional note here, having consistent naming convention for env variables is very important. We should ensure that we should make this a part of the documentation standards.

File: demoScripts/createVMAndSSH.md
Comment (rgardler-msft) [inline demoScripts/createVMAndSSH.md:17]: No command block

File: demoScripts/createVMAndSSH.md
Comment (rgardler-msft) [inline demoScripts/createVMAndSSH.md:47]: Lovely seeing multiline support :-)

File: demoScripts/createVMAndSSH.md
Comment (rgardler-msft) [inline demoScripts/createVMAndSSH.md:79]: This isn't testing an actual connection, nor does the being shown in the doc create a connection.

File: testScripts/createAKSFile.md
Comment (rgardler-msft) [inline testScripts/createAKSFile.md:4]: Huh? if it is fixed why is it still here? Is this a note for the reviewer that it is fixed in a PR submitted after this one? If so then make this clearer, preferably with a PR #

File: testScripts/createAKSFile.md
Comment (rgardler-msft) [inline testScripts/createAKSFile.md:23]: Odd indenting

File: testScripts/createAKSFile.md
Comment (rgardler-msft) [inline testScripts/createAKSFile.md:115]: Why?

I'm stopping the review here, I'm not sure what the point of this file is and the beginning says it is broken anyway.

Tests need to be readable and meanigful.  If the goal is to demonstrate the break then it should be in a working branch for that break. Create an issue, create a branch commit the test to that branch, fix it, merge that branch.

**Review by rgardler-msft — CHANGES_REQUESTED** [review]
Some of these comments are about the quality of the human readable content. I'd hope we can improve the content as we work on existing docs, but that should be coordinated with the docs team and thus these comments are not blockers.

There are some comments on how Innovation Engine is being used here. These do need to be addressed before merge OR we need to agree the current implementation is sub-optial or not and when appropriate create issues for those items.

File: demoScripts/aksQuickStart.md
Comment (jasonmesser7) [reply demoScripts/aksQuickStart.md:22]: Innovation Engine currently does not do anything with Includes. We would need to add to the parser to pull out this information from Paragraphs and then we could do something with it. We should create an issue to add Include functionality.

File: demoScripts/aksQuickStart.md
Comment (jasonmesser7) [reply demoScripts/aksQuickStart.md:40]: I do not test/verify success of every command. I agree that a north star should be to test every single command. However, given time I am primarily only testing the end result. I am making the assumption that if we get to the end of the tutorial and an AKS cluster is deployed with a working website that this step also succeeded.

File: demoScripts/aksQuickStart.md
Comment (jasonmesser7) [reply demoScripts/aksQuickStart.md:50]: Good point - this is something we should take up with the Docs team

File: demoScripts/aksQuickStart.md
Comment (jasonmesser7) [reply demoScripts/aksQuickStart.md:56]: I think it makes sense to make naming variables less generic. Should we also put a random identifier at the end in case someone wants to run the tutorial multiple times?

File: demoScripts/aksQuickStart.md
Comment (jasonmesser7) [reply demoScripts/aksQuickStart.md:61]: This works in the actual Azure docs QuickStart with the way their content is organized. https://github.com/MicrosoftDocs/azure-docs/blob/main/articles/aks/learn/quick-kubernetes-deploy-cli.md

File: demoScripts/aksQuickStart.md
Comment (jasonmesser7) [reply demoScripts/aksQuickStart.md:66]: Good catch - When merging a document earlier with Cynthia into Azure docs she just removed this
```suggestion

```

File: demoScripts/aksQuickStart.md
Comment (jasonmesser7) [reply demoScripts/aksQuickStart.md:75]: Ack on thinking about a best practice. Because of all the differences, these tend to come out lower than I would hope. For example, here is one for VM create that was just today which failed.
image

File: demoScripts/aksQuickStart.md
Comment (jasonmesser7) [reply demoScripts/aksQuickStart.md:91]: Lets bring this up with the AKS docs team / PG
```suggestion
Create an AKS cluster using the [az aks create][az-aks-create] command with the `--enable-addons monitoring` and `--enable-msi-auth-for-monitoring` parameter to enable [Azure Monitor Container insights][azure-monitor-containers] with managed identity authentication (preview). The following example creates a cluster with one node and enables a system-assigned managed identity:
```

File: demoScripts/aksQuickStart.md
Comment (jasonmesser7) [reply demoScripts/aksQuickStart.md:100]: Yes - it will be a random identifier added to the end of the original name

File: demoScripts/aksQuickStart.md
Comment (jasonmesser7) [reply demoScripts/aksQuickStart.md:94]: See logic above for not testing every command, but I agree that this would be an important one to test.

File: demoScripts/aksQuickStart.md
Comment (rgardler-msft) [reply demoScripts/aksQuickStart.md:22]: OK I will raise an issue, no need to block on this. Not sure what the priority is at this point.

File: demoScripts/aksQuickStart.md
Comment (jasonmesser7) [reply demoScripts/aksQuickStart.md:109]: I agree that checking before is better. I got pushback in my original E2E AKS deployment from the field and so I just went back to this basic command as it doesn't do any harm to try and re install it. 

In your opinion is there any reason to sacrifice readability?

File: demoScripts/aksQuickStart.md
Comment (jasonmesser7) [reply demoScripts/aksQuickStart.md:241]:
```suggestion
    * To do this programmatically via the command line run the following command:
```

File: demoScripts/aksQuickStart.md
Comment (rgardler-msft) [reply demoScripts/aksQuickStart.md:40]: I feel this approach is suboptimal. To show the true value of this testing framework it is not sufficient to say "it failed". We need to tell the docs team, who are responsible for all docs and SMEs in none, why it failed and where. This will often allow them to fix it without engaging a SME. If they do need to engage the SME they will be able to provide the error output from Innovation Engine which, more often than not, will be enough for the SME to identify the fix. Thus making the process significantly more efficient.

Furthermore, results sections make the content more easy to follow for human readers as they can proceed with confidence that the output they got was correct.

File: demoScripts/aksQuickStart.md
Comment (jasonmesser7) [reply demoScripts/aksQuickStart.md:243]: Agreed. Let's chat with docs and see the best way that we can handle this. IMO if all of the rest of the commands are via CLI then we can make this one too. A simple copy paste is easier instead of having the user create a file locally, open it, come back and hit copy, then paste it in. I suspect this will get some resistance here though.

File: demoScripts/aksQuickStart.md
Comment (rgardler-msft) [reply demoScripts/aksQuickStart.md:50]: I don't see that this is a docs issue. If the command does not work unless it has sudo privileges then it's a CLI or an Innovation Engine issue, isn't it?

File: demoScripts/aksQuickStart.md
Comment (rgardler-msft) [reply demoScripts/aksQuickStart.md:56]: For readability I would avoid a random postfix. In other reviews I've said we need to explain why this content is here. So perhaps we should have a boilerplace that includes the reasoning and gives an example of how to add a random postfix if desired.

File: demoScripts/aksQuickStart.md
Comment (jasonmesser7) [reply demoScripts/aksQuickStart.md:357]: I like that idea. However, doesn't that break one of our rules? This was from chat the other day - 

"I'm uncomfortable with the proposal to ignore it. It breaks the rule that we should not have a different behavior in the engine vs standalone execution. This is a very important rule and should always be enforced. Failure to do so means that we are no longer testing the thing that is going into the customers hands."

Are we okay with altering behavior during testing?

File: demoScripts/aksQuickStart.md
Comment (jasonmesser7) [reply demoScripts/aksQuickStart.md:373]: Totally agree - that is a good catch. I thought I validated the output but that must have been on a separate tutorial.

File: demoScripts/aksQuickStart.md
Comment (jasonmesser7) [reply demoScripts/aksQuickStart.md:386]: It is cleaned up, but that would be good to specify.

File: demoScripts/createVMAndSSH.md
Comment (jasonmesser7) [reply demoScripts/createVMAndSSH.md:17]: This is another one to figure out how to handle, we are not able to do a successful Az login within Innovation engine. As so, taking this out of a command block is ideal. (This was how the document was already written)

File: testScripts/createAKSFile.md
Comment (jasonmesser7) [reply testScripts/createAKSFile.md:115]: We can remove this line. The final echo was part of the testing process. The beginning of this test was meant to explain the test. There was an issue with indentation when creating YAML files. I added the note at the top to say that this is no longer an issue, but I wanted to leave the description of the issue solved to show the history. I can do this easily by just rewriting the initial description.

File: demoScripts/createVMAndSSH.md
Comment (jasonmesser7) [reply demoScripts/createVMAndSSH.md:79]: Good call - I did this as Innovation Engine does not support interactive SSH commands. The idea here was that it would echo this and then exit the program and the user could simply copy the output and run SSH.

We should do something like this though to validate it works

ssh $MY_USERNAME@$IP_ADDRESS whoami

**Pull request closed without merging**
